lib/linter/linter.js

```diff
@@ -1694,6 +1663,7 @@ class Linter {
                 slots.lastSourceCode = new SourceCode({
                     text: slots.lastSourceCode.text,
                     ast: slots.lastSourceCode.ast,
+                    hasBOM: slots.lastSourceCode.hasBOM,
```

Comment:
We should probably pass `hasBOM: slots.lastSourceCode.hasBOM` here as well:

https://github.com/eslint/eslint/blob/389744be255717c507fafc158746e579ac08d77e/lib/linter/linter.js#L1396-L1402

It seems this was a bug that this property was lost in the copy.